src/pudl/io_managers.py

```diff
@@ -349,13 +356,26 @@ def load_input(self, context: InputContext) -> pd.DataFrame:
                 name.
         """
         table_name = self._get_table_name(context)
+        pkg = Package.from_resource_ids()
+        all_resources = [resource.name for resource in pkg.resources]
+        res = None
+        if table_name in all_resources:
+            res = pkg.get_resource(table_name)
+
```

Comment:
@bendnorman I'm still a little unclear on what our expectations are about tables that would get written to SQLite with this IOManager.

Without the check to see whether the table had a resource, some unit tests failed, since they just had a temporary table with no metadata associated with it.

Do we only expect to write tables to SQLite that have `Resource` classes defined? Or do we want to be able to write interim tables that are so stringently described?  If so, should we differentiate between those cases somehow in the IO Manager?

Reply:
I think we should expect all tables we are loading from the database to have a Resource class defined so we can be confident we are working with the correct dtype information.

`self._get_sqlalchemy_table(table_name)` throws an error if the table doesn't exist in the metadata.